filebeat/harvester/log_file.go

```diff
@@ -167,9 +165,6 @@ func (r *LogFile) wait() {
 }
 
 func (r *LogFile) Close() {
-	// Make sure reader is only closed once
-	r.singleClose.Do(func() {
-		close(r.done)
-		// Note: File reader is not closed here because that leads to race conditions
-	})
+	close(r.done)
```

Comment:
As far as I can see there is now only one place left where the fileReader is closed and that is when the goroutine above stops. It seems the Once was still in for historical reason where filereader was closed in different places (initially we had 2 go routines, one for the h.done channel and one for the timeout.